### PR TITLE
Feature/travis integration

### PR DESCRIPTION
**Pull request by mrothauer**

### CODE REVIEW

**Comment by mrothauer** [comment]
killer feature :-) all tests run automatically with travis ci integration
tests are executed on push and creating PF and if .travis file exists
please do not merge - i want to clean the commit mess i needed for testing before that

**Comment by k-pd** [comment]
What do You expect me to review?

**Comment by mrothauer** [comment]
maybe .travis.yml can be implemented simpler? i'm not an expert in shell scripting, bash, sudo and so on :-)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+services:
+  - mysql
+  
+env:
+  - DB=mysql
+  
+before_script:
+  # email mock
+  - sudo apt-get install -y -qq postfix
```
Comment (mrothauer) [inline .travis.yml]: this line and the follwing lines are for preventing errors when emails are sent from the application.
http://michaelthessel.com/mock-mail-setup-for-travis-ci/

**Comment by mrothauer** [comment]
* hostname... i don't understand what you mean. everything seems to work :-) maybe entry in hosts file?
* just added php 5.5 and php 7.0
* php-fpm: just copy/pasted it from https://docs.travis-ci.com/user/languages/php/
"Currently Travis CI does not support mod_php for apache, but you can configure php-fpm for your integration tests."

**Comment by k-pd** [comment]
postfix: OK.
PHP5.5/7.0: OK.
PHP-FPM: OK.

Hostname: I understand, that the tests are passing somehow. But I don't understand, why simplebrowser can successfully resolve `www.foodcoopshop.dev` without adding that domain name to /etc/hosts. I would prefer to have that added explicitly to ensure the tests continue to work if Travis desides to change that (quite odd) VM behaviour.

**Comment by k-pd** [comment]
Another question: How can I use that testing facility from my repo before pushing to GitHub?

**Comment by mrothauer** [comment]
* hosts: you're right, i added it to hosts
* no idea - now you know why i made so many pushes :-) maybe installing the travis client helps? https://github.com/travis-ci/travis.rb

**Comment by k-pd** [comment]
This Travis client thingy is only for commandline usage of the Travis CI Website. No option to use anything but GitHub repositories for testing. There's some instructions around the internet for "debugging travis ci trouble locally" which is mostly what I was thinking of. Maybe I find time to make that available as a pre-push git hook...

**Pull request merged**